cli/install/install.go

```diff
@@ -1004,17 +1004,48 @@ func installTarantool(binDir string, incDir string, installCtx InstallCtx,
 	}
 
 	// Check if program is already installed.
+	// If it is installed, check if newest version exists.
 	if !installCtx.Reinstall {
 		log.Infof("Checking existing...")
 		if util.IsRegularFile(filepath.Join(binDir, versionStr)) &&
 			util.IsDir(filepath.Join(incDir, versionStr)) {
-			log.Infof("%s version of tarantool already exists, updating symlinks...", versionStr)
-			err := changeActiveTarantoolVersion(versionStr, binDir, incDir)
+			isBinExecutable, err := util.IsExecOwner(filepath.Join(binDir, versionStr))
 			if err != nil {
 				return err
 			}
-			log.Infof("Done")
-			return err
+
+			curBinVersion := version.Version{}
+			var lastCommitHash string
+
+			if isBinExecutable {
+				tarantoolBin := cmdcontext.TarantoolCli{
+					Executable: filepath.Join(binDir, versionStr),
+				}
+				curBinVersion, err = tarantoolBin.GetVersion()
+				if err != nil {
+					return err
+				}
+				curBinVersion.Hash = curBinVersion.Hash[1:]
+
+				lastCommitHash, err = getCommit(installCtx.Local, distfiles,
+					search.ProgramCe, tarVersion)
+				if err != nil {
+					return err
+				}
+			}
+
+			if curBinVersion.Hash == lastCommitHash[:len(curBinVersion.Hash)] {
```

Comment:
The condition confusing, it will be obvious:
```suggestion
			if !isBinExecutable || strings.HasPrefix(lastCommitHash, curBinVersion.Hash) {
```

Reply:
Changed this condition.